torch/_dynamo/decorators.py

```diff
@@ -58,15 +72,35 @@ def disable(fn=None, recursive=True):
     but still process recursively invoked frames.
     """
     if recursive:
+        ctx = (
+            OldDisableContext()
+            if _old_compile_disable_behavior()
+            else CompileEnabledContext(False)
+        )
         if fn is not None:
-            fn = innermost_fn(fn)
+            if _old_compile_disable_behavior():
+                fn = innermost_fn(fn)
             assert callable(fn)
-            return DisableContext()(fn)
-        return DisableContext()
+            return ctx(fn)  # type: ignore[operator]
+        return ctx
     else:
         return skip(fn)
 
 
+def enable(fn=None):
+    """
+    Decorator to re-enable TorchDynamo - inverse of `disable`.
+
+    Compilation will only occur if there was a previous `compile` call.
```

Comment:
Are your canonical docs going to live here or in torch.compiler?

Reply:
torch.compiler